From 17a49c0a29704b3db79680dc3f9ac0cca1403eff Mon Sep 17 00:00:00 2001
From: QbProg <tholag@gmail.com>
Date: Sat, 11 Jun 2011 15:27:45 +0200
Subject: [PATCH 14/26] [warning-fix][const_standard_cstring]

Dealing with some const warnings revealed when using gcc 4.5.2

(moved from hs/const-string-warnings branch)

Note: this commit breaks API compatibility
---
 inc/FSD_File.hxx                               |    2 +-
 inc/TCollection_ExtendedString.hxx             |    2 +-
 src/FSD/FSD_File.cxx                           |    2 +-
 src/TCollection/TCollection_ExtendedString.cxx |    2 +-
 4 files changed, 4 insertions(+), 4 deletions(-)

Index: oce/inc/FSD_File.hxx
===================================================================
--- oce.orig/inc/FSD_File.hxx
+++ oce/inc/FSD_File.hxx
@@ -319,7 +319,7 @@ protected:
 private:
 
   
-  Standard_EXPORT   static const Standard_CString MagicNumber() ;
+  Standard_EXPORT   static Standard_CString MagicNumber() ;
 
 
 FSD_FStream myStream;
Index: oce/inc/TCollection_ExtendedString.hxx
===================================================================
--- oce.orig/inc/TCollection_ExtendedString.hxx
+++ oce/inc/TCollection_ExtendedString.hxx
@@ -248,7 +248,7 @@ friend Standard_EXPORT Standard_OStream&
 //!    aString.Token("; :,",2) returns "test" <br>
   Standard_EXPORT     TCollection_ExtendedString Token(const Standard_ExtString separators,const Standard_Integer whichone = 1) const;
   //! Returns pointer to ExtString <br>
-  Standard_EXPORT    const Standard_ExtString ToExtString() const;
+  Standard_EXPORT    Standard_ExtString ToExtString() const;
   //! Truncates <me> to <ahowmany> characters. <br>
 //! Example:  me = "Hello Dolly" -> Trunc(3) -> me = "Hel" <br>
 //!   Exceptions <br>
Index: oce/src/FSD/FSD_File.cxx
===================================================================
--- oce.orig/src/FSD/FSD_File.cxx
+++ oce/src/FSD/FSD_File.cxx
@@ -136,7 +136,7 @@ Storage_Error FSD_File::Close()
 //purpose  : ------------------ PROTECTED
 //=======================================================================
 
-const Standard_CString FSD_File::MagicNumber()
+Standard_CString FSD_File::MagicNumber()
 {
   return MAGICNUMBER;
 }
Index: oce/src/TCollection/TCollection_ExtendedString.cxx
===================================================================
--- oce.orig/src/TCollection/TCollection_ExtendedString.cxx
+++ oce/src/TCollection/TCollection_ExtendedString.cxx
@@ -974,7 +974,7 @@ TCollection_ExtendedString TCollection_E
 // ----------------------------------------------------------------------------
 // ToExtString
 // ----------------------------------------------------------------------------
-const Standard_ExtString TCollection_ExtendedString::ToExtString() const
+Standard_ExtString TCollection_ExtendedString::ToExtString() const
 {
   if(mystring) return mystring;
   return NULL_EXTSTRING;
